Infotip for small screens

raveenakothapally opened the pull request:
# InfoTips
Infotips activated when engaged by the user. Tapping or clicking the info bubble initiates flyout. On small screens, it's a modal.

![Kapture 2020-05-06 at 9 18 39](https://user-images.githubusercontent.com/23395619/81201878-a7aaeb80-8f7a-11ea-9165-8590d580b035.gif)

agliga commented on src/less/tooltip/base/tooltip.less:
```diff
@@ -162,6 +162,66 @@ span.tourtip {
+.dialog-infotip__close {
+    background-color: @dialog-back-close-background-color;
+    border: 0;
+    border-top-right-radius: 7px;
```
If we change the `margin` to be greater than `7px`, then this isn't needed. But then the `padding` will need to be reduced to compensate
So maybe `padding: 8px; margin: 8px;` works

raveenakothapally replied:
Thanks for the tip!

agliga commented on src/less/tooltip/base/tooltip.less:
```diff
@@ -162,6 +162,66 @@ span.tourtip {
+    }
+}
+
+.dialog-infotip__window:not(.dialog-infotip__window--full) {
```
This is a weird way to do this. Can you not do the `dialog-infotip__window--full` as a separate class and have those styles there? 
So the base should have all the styles it needs, and the full should have what it needs (and override if needed). 
Remember in CSS order matters to in terms of overriding styles. 

Also there is no full example

agliga commented on src/less/tooltip/base/tooltip.less:
```diff
@@ -162,6 +162,66 @@ span.tourtip {
+    .dialog-window();
+
+    padding-bottom: 15vh;
+    top: 15%;
```
This is getting overriden by the `top:50` in the the other styles, is it needed?

agliga commented on src/less/tooltip/base/tooltip.less:
```diff
@@ -162,6 +162,66 @@ span.tourtip {
+    .dialog-body-content();
+
+    margin-right: 0;
+    min-height: 18px;
```
Why is the `min-height` needed? I removed it and was working fine.

raveenakothapally commented on src/less/tooltip/base/tooltip.less:
```diff
@@ -162,6 +162,66 @@ span.tourtip {
+    left: 50%;
+    max-width: 400px;
+    min-width: 200px;
+    top: 50%;
+    transform: translate(-50%, -25%);
```
To center the dialog horizontal and vertical

agliga replied:
So that's fine, I'm just wondering why it's `-25%` and not `-50%` to center it 
https://css-tricks.com/centering-css-complete-guide/ (In the centering both vertically and horizontally and unknown width/height)

raveenakothapally replied:
I used that same guideline but `-50%` didn't exactly center it :)

raveenakothapally commented:
> I looked at it purely form a code/functional point of view. I did not look at the spec and see how well it matches.
> Overall nice job, I think the main comments were mainly nits, but the overall code looks great and is exactly what I was envisioning.
> I almost feel like it belongs as a separate module although I'll let Ian make that final call.

Thanks for the review @agliga. Addressed your concerns; could you please take another look?
The left and transform is done in order to center the dialog horizontally and vertically (Spec wants this to be centered). Please let me know if you have any other suggestions to achieve this behavior.

ianmcburnie commented on docs/_includes/common/tooltip.html:
```diff
@@ -181,6 +181,33 @@ <h3 class="infotip__heading">Infotip</h3>
 </p>
     {% endhighlight %}
 
+    <p>Infotip on small screens</p>
```
@raveenakothapally Let's also add this demo and code example under the dialog section. Before release I will most likely rename it something a little more generic than `dialog-infotip`, because I see it's potential in places other than infotip, but the current name will suffice for now.

ianmcburnie commented on src/less/tooltip/base/tooltip.less:
```diff
@@ -162,6 +162,58 @@ span.tourtip {
     width: @icon-width;
 }
 
+.dialog-infotip[role="dialog"] {
```
We should move all of this `dialog-infotip` under the dialog module. Infotip will then have a dependency on the dialog module.

ianmcburnie commented on dist/drawer/ds6/drawer.css:
```diff
@@ -91,10 +91,6 @@
 .drawer__window {
   display: -webkit-box;
   display: flex;
-  -webkit-box-orient: vertical;
```
Is this change intentional? Looks like it has broken the drawer:

<img width="577" alt="Screen Shot 2020-05-05 at 9 05 33 PM" src="https://user-images.githubusercontent.com/38065/81137711-929b7180-8f14-11ea-9288-dfd41e0ca98d.png">

raveenakothapally replied:
I searched for `.dialog-window()` and it did not give a usage in drawer (As it uses `.dialog-window(100%)`) so missed updating `drawer` module. Thanks for catching this bug. This should be fixed now.

ianmcburnie commented on src/less/dialog/base/dialog.less:
```diff
@@ -186,12 +188,63 @@
+    background-color: @dialog-back-close-background-color;
+    border: 0;
+    color: @dialog-back-close-color;
+    cursor: pointer;
```
Nitpick, but please remove `cursor: pointer`. For a11y reasons we only have pointer on links, not buttons. Also I don't think `z-index` is needed...?

raveenakothapally replied:
ACK on `cursor`

`z-index` is something that is copied over from `dialog__close`. Not sure why it was needed back during dialog impl. Could be to give precedence (Top of) over default elements just in case.

ianmcburnie commented:
Going with #1100 instead. Closing.
